### ISSUE TITLE
Problem of foot skating.

### ISSUE THREAD

**Issue by Wenretium** [Mathux/TEMOS#37]
Hello! First of all, thanks for your excellent work. I ran your interact code and tested it on all three of the pre-trained models provided, but found that the results had **severe foot skating**. It was different from the effects described in the paper.
These are the results of the text "A person takes two steps, raises his right hand, keeps walking" on the three models. Problems also arise with other text inputs.

[a_person_takes_two_steps,_raises_his_right_hand,_keeps_walking_len_60.webm](https://github.com/Mathux/TEMOS/assets/55617704/88657b25-50cb-42a9-a753-2cad6c6167a2)

[a_person_takes_two_steps,_raises_his_right_hand,_keeps_walking_len_61.webm](https://github.com/Mathux/TEMOS/assets/55617704/b8642b3c-d31d-4c47-aa75-4786b461c5d8)

[a_person_takes_two_steps,_raises_his_right_hand,_keeps_walking_len_62.webm](https://github.com/Mathux/TEMOS/assets/55617704/6efb285b-ae54-49f3-a992-ac850f2d32bd)

**Comment by Mathux** [comment]
Hello @Wenretium,

I think the problem arise because of a mismatch between training and testing distributions. Most of the text of KIT-ML consists of a text with a unique action "A person walks forward" or "A person is sitting down". In your example, you are asking for 3 consecutive actions "takes two steps" + "raises his right hand" + "keeps walking", with a duration of 5 seconds. 

For doing consecutive actions, you can look at the work TEACH: https://teach.is.tue.mpg.de

**Comment by Wenretium** [comment]
Thanks for your quick reply! I also tested on "A human plays basketball", which is one of the examples in the paper. Sometimes people's feet seem to float on the ground. Have you ever encountered this unreal phenomenon?

[a_human_plays_basketball_len_120.webm](https://github.com/Mathux/TEMOS/assets/55617704/96cbeda9-3675-45a9-8c37-d3010ce02bfe)

[a_human_plays_basketball_len_100.webm](https://github.com/Mathux/TEMOS/assets/55617704/f981cb9c-5d62-4809-ae62-e3cd8a2301c5)

**Comment by Mathux** [comment]
Yes indeed, this generated motion does not look right. I already encountered this but not often. 

If you really care about foot sliding etc, you can add an extra foot sliding loss during training or you could use a physic simulator to correct the motion.
